Automattic/wp-calypso
# Store: Add basic review listing screen

This PR implements a listing view based off of the design [here](https://automattic.invisionapp.com/share/R4BQDYYVY#/screens/251026936). It allows for filtering by status, has pagination, and allows you to search all reviews for a search term.

@kellychoffman @jameskoster This PR does have some CSS, however I expect some of the structure to change a bit as I start to implement the expanded/editor state of the reviews. Just a heads up. My main goal for this PR was to get the functionality working.

<img width="819" alt="screen shot 2017-09-08 at 3 37 19 pm" src="https://user-images.githubusercontent.com/689165/30233911-9addf228-94ad-11e7-89be-b3395ce51204.png">
<img width="845" alt="screen shot 2017-09-08 at 3 38 14 pm" src="https://user-images.githubusercontent.com/689165/30233909-9ad7211e-94ad-11e7-901f-7e5d48c58f39.png">
<img width="794" alt="screen shot 2017-09-08 at 3 37 12 pm" src="https://user-images.githubusercontent.com/689165/30233910-9ad81736-94ad-11e7-9e13-6fac6f78186d.png">

Closes #17042.

To Test:
* Run `npm run test-client client/extensions/woocommerce/` and make sure all tests pass.
* You **must** use my `wc-api-dev` branch and not the production copy of `wc-api-dev`. Here is a copy: https://cloudup.com/cOW3IXvOy1u. Just download the zip, and upload it via the `wp-admin` plugin installer. You can do this on AT sites.
* Make sure you have a few reviews on your site. You can import the demo content, or post a few.
* Load up this branch.
* Go to `http://calypso.localhost:3000/store/reviews/:site` and verify that you see the filter/search bar.
* Select a tab that has reviews. Verify that the reviews show up correctly.
* Select a tab that has no reviews. Verify that you get an EmptyContent message.
* Enter a search term. You should see some results show up, or a EmptyContent message.
* (Optional) To test pagination if you don't have a ton of reviews, you can update  the `perPage` number in `reviews-list.js`, and the 'per_page' number in `/state/sites/reviews/utils.js`.
